### DISCUSSION TITLE
Question about auth design / TypeScript and session interface

### DISCUSSION THREAD

**Discussion by oscarbatori** [nextauthjs/next-auth#10223] [Help] [answered]
I am using AuthJS v5 on a new project. The setup is as follows:
- NextJS app for frontend
- Python API server
- Auth0 as my provider

The integration is working nicely where the NextJS application handles the authentication, which provides an access token which is in turn presented to the API to authenticate requests. My question is about some TS compiler errors, and what they imply about the decision to use callbacks to handle API interactions that are part of the login flow:
- Putting the `accessToken` on the `session` is causing TypeScript to complain, but I don't see a way to override the session definition without implementing an adapter?
- Is this a case where I really should be using an adapter, or need to if I want to mollify the TS compiler? Handling user creation in the callbacks does feel a _bit_ icky, but I am completely new to web dev so I am not inclined to trust my judgment here.

I am early in my product development and going with Auth0 only based login for simplicity (basically leaning on that service for account management), so I would prefer to avoid implementing an adapter, which doesn't seem incredibly well document for the case when the backend is an API, not a DB.

Thanks in advance for any advice!

**Comment by mvn-h24** [comment, marked as answer]
> * but I don't see a way to override the session definition without implementing an adapter?

Here is the answer https://next-auth.js.org/getting-started/typescript#module-augmentation
Your need to customize `Session` interface and provide path to this custom .d.ts file in tsconfig.json in `include` array (like `next-env.d.ts` file)

**Reply by oscarbatori** [reply]
Oh stellar, thanks for this. I've been a bit confused kind of piecing stuff together across v4 / experimental docs (totally understandable given state of the project!).
